### PR TITLE
Move portable facade validation

### PR DESCRIPTION
**Pull request by jaredpar**
This moves the portable facade validation into BuildBoss. This has two
advantages:

1. Centralizes our logic for validating the correctness of our SWR files
2. Uses the same approach for validating SWR files in all cases: build output vs.
inspecting project.assets.json. The former is more robust as it's
examining build output directly vs. projecting build output based on the
contents of project.assets.json

### CODE REVIEW

File: src/Tools/BuildBoss/CompilerNuGetCheckerUtil.cs [hunk 1]
```diff
@@ -25,15 +25,15 @@ namespace BuildBoss
     /// The least crazy solution that could be decided on was to manage the list of dependencies 
     /// by hand in the NuSpec file and then rigorously verify the solution here.
     /// </summary>
-    internal sealed class CompilerNuGetCheckerUtil : ICheckerUtil
+    internal sealed class PackageContentsChecker : ICheckerUtil
```
Comment (jaredpar) [inline src/Tools/BuildBoss/CompilerNuGetCheckerUtil.cs]: The file will be renamed to match in a later PR. Wanted to keep this easy to review.

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure, @tmeschter for review

File: src/Tools/BuildBoss/CompilerNuGetCheckerUtil.cs [hunk 2]
```diff
@@ -75,18 +76,25 @@ private bool CheckDesktop(TextWriter textWriter)
...
+            var swrRelativeFileNames = FilterRelativeFileNames(
+                dllRelativeNames,
+                "Microsoft.DiaSymReader.Native.amd64.dll",
+                "Microsoft.DiaSymReader.Native.x86.dll").ToList();
 
-            allGood &= VerifySwrFile(textWriter, dllRelativeNames);
+            // The Microsoft.CodeAnalysis.Compilers.swr file is used in part to ensure NGEN is run on the set of 
+            // facades / implementation DLLs the compiler depends on. This set of DLLs is the same as what is 
+            // included in our NuGet package. Need to make sure all the necessary managed DLLs are included here.
+            // TODO: fix this
```
Comment (jasonmalinowski) [inline src/Tools/BuildBoss/CompilerNuGetCheckerUtil.cs]: TODO comment seems...unclear. What needs fixing?

Comment (jaredpar) [reply src/Tools/BuildBoss/CompilerNuGetCheckerUtil.cs]: Narrator: he saw that comment and fixed the issue but left the comment because ...

**Comment by jaredpar** [comment]
Ping @dotnet/roslyn-infrastructure for review

**Pull request merged**